doc/BGP/BGP-route-aggregation-with-bbr-awareness.md

```diff
@@ -0,0 +1,349 @@
+    - [Bgp Container Behavior](#bgp-container-behavior)
+        - [UML Sequence Diagrams](#uml-sequence-diagrams)
+    - [CLI Design](#cli-design)
+        - [Show CLI](#show-cli)
+        - [Config CLI](#config-cli)
+
+
+
+## Revision
+
+| Revision | Date        | Author           | Change Description |
+| -------- | ----------- | ---------------- | ------------------ |
+| 1.0      | Jul 17 2024 | Wenda Chu, Jing Kan | Initial proposal   |
+
+## Definitions/Abbreviations
+
+| Definitions/Abbreviation | Description |
+| ------------------------ | ----------- |
+| BGP | Border Gateway Protocol |
+| FRR | A free and open source Internet routing protocol suite for Linux and Unix platforms |
+| BBR | Full name is Bounce Back Routing, a feature that can prevent packet drop caused by aggregated address.|
+
+
+## Scope
+
+This document describes how to leverage the SONiC config DB to add or remove BGP aggregate address and the SONiC state DB to track the state of it with BBR awareness
+
+
+## Overview
+In BGP, we can aggregate contributing routes into one single aggregated route. It has many advantages, for example reducing routes’ count.
+
+However, firstly, SONiC can’t configurate aggregated addresses via config DB and doesn’t have CLI support for it.
+
+Secondly, if we aggregated routes without BBR feature on device, we many got packet drop on this device due to contributing routes missing.
+
+To leverage the benefit of address aggregation and the BBR feature, we are trying to design the aggregate address configuration mechanism with BBR awareness in this doc.
+
+
+## Requirements
+With this design, users can add or remove aggregated address via editing config DB or config CLI with parameters to control the aggregation and route announcement behavior and can use the show command to check aggregated address on device.
+
+With BBR awareness, aggregated address will be generated conditionally depend on BBR feature's state.
+
+## High Level Design
+First we introduce the config DB extension which define the feature scope and parameters we have.
+Then we introduce how the bgp container will change its behavior accordingly.
+
+### Config DB Extension
+We define a new YANG model to add a new key named `BGP_AGGREGATE_ADDRESS` in config DB.
+The key will index to a list of aggregated addresses with their parameters.
+
+The YANG model and config DB demo are showed in below:
+
+#### Yang Model
+```
+module sonic-bgp-aggregate-address {
+
+    namespace "http://github.com/sonic-net/sonic-bgp-aggregate-address";
+
+    prefix bgp-aggregate-address;
+
+    yang-version 1.1;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONIC BGP aggregate address configuration module.";
+
+    revision 2024-07-17 {
+        description
+            "Initial revision.";
+    }
+
+    container sonic-bgp-aggregate-address {
+        container BGP_AGGREGATE_ADDRESS {
```

Comment:
I think, you are planning to handle aggregate address configs via bgpcfgd, we should mention that clearly in the YANG models, we do have another parallel config handler via frrcfgd, the community should NOT be confused with what model goes to which handler. May be creating a folder under https://github.com/sonic-net/sonic-buildimage/tree/master/src/sonic-yang-models/yang-models  e.g bgpcfgd to have all the YANG models relevant for it and frrcfgd folder for the models applicable for the frrcfgd would clear confusion.

router bgp 1
 address-family ipv4 unicast
  aggregate-address 10.0.0.0/8
  aggregate-address 20.0.0.0/8 as-set
  aggregate-address 40.0.0.0/8 summary-only
  aggregate-address 50.0.0.0/8 route-map aggr-rmap
 exit-address-family

Reply:
Thanks, I will mention that it is dedicated for bgpcfgd in YANG model.

I am not sure if we create a new folder that won't break anything, you can open an issue to discuss this.